### ISSUE TITLE
Pictures app very slow with hig number of pictures

### ISSUE THREAD

**Issue by patchlog** [owncloud/apps#790]
The Pictures app is very slow when you have thousands of pictures.

It downloads a list of all images right when you enter the Pictures with a call to getimages.php. Mine was 15 mb of data and made me wait about 1 minute just to get to the top screen with only 4 albums.

Why do we need all of this data from the beginning?

**Comment by oparoz** [comment]
As discussed here: https://github.com/interfasys/galleryplus/issues/17 , loading only images for the current folder also has its downsides.

**Comment by oparoz** [comment]
First step, stop loading all images.
Available in this branch: https://github.com/interfasys/galleryplus/tree/performance-fixes

```
# git clone -b performance-fixes https://github.com/interfasys/galleryplus.git
```

**Comment by leandro-lucarella-sociomantic** [comment]
> will reorganizing a flat gallery into nested folders help?

No, I have thousands of images all organized in a hierarchical tree, with very few pictures in each actual leaf folder. My gallery never ends to load.

**Comment by leandro-lucarella-sociomantic** [comment]
Gallery+ works nicely for me, thanks!

**Comment by oparoz** [comment]
@leandro-lucarella-sociomantic - Ah, cool. You might run into issues as you reach your leaf folders, but that should be fixed in V2.1. I just had to rewrite a lot of the Javascript to make it ID based vs Path based and also to try and make the view less fragile.

**Comment by leandro-lucarella-sociomantic** [comment]
On Thu, Feb 26, 2015 at 07:59:44AM -0800, Olivier Paroz wrote:
> @leandro-lucarella-sociomantic - Ah, cool. You might run into issues
> as you reach your leaf folders, but that should be fixed in V2.1. I
> just had to rewrite a lot of the Javascript to make it ID based vs
> Path based and also to try and make the view less fragile.

It's working fine so far, thanks again!

**Comment by GreatEmerald** [comment]
So are there plans to include the updated app into the base ownCloud?

**Comment by oparoz** [comment]
@GreatEmerald - Yes indeed. It will be part of 8.2. You can use it now for 8.1 and 8.0 if you install Gallery from the Multimedia section.
